src/Models/Laravel/Token.php

```diff
@@ -154,7 +154,7 @@ protected static function newFactory(): TokenFactory
     protected function pivotIdentifier(): Attribute
     {
         return Attribute::make(
-            get: fn () => $this->token_chain_id,
+            get: fn () => "{$this->collection->collection_chain_id}:{$this->token_chain_id}",
```

Comment:
I think it's better to do null coalesce incase collection is not pulled
```suggestion
            get: fn () => "{$this->collection?->collection_chain_id}:{$this->token_chain_id}",
```

Reply:
This should never happen, I suggest throwing an error if the collection doesn't get pulled as that would mean something is broken

Reply:
I think its possible to happen even if collection exist in db, its when we didn't eager load the `collection` relation

Reply:
Then this needs to be refactored, the pivot is supposed to be a unique identifier we can't have an invalid identifier it defeats the whole purpose of being an identifier.

Reply:
We would definitely need to make sure we get the collection id for this to work, unless we have any other on-chain data available we can use to make the identifier unique and discoverable after a sync.